tensorflow/core/ops/dataset_ops.cc

```diff
@@ -399,6 +399,22 @@ batch_size: A scalar representing the number of elements to accumulate in a
   batch.
 )doc");
 
+REGISTER_OP("SlideDataset")
+    .Input("input_dataset: variant")
+    .Input("slide_size: int64")
```

Comment:
Naming nit: I think `window_size` would be less ambiguous. (Is the "slide size" the size of the window or the size of the slide from one window to the next?)